dialog/receiver.go

```diff
@@ -23,6 +23,46 @@ import (
 	"github.com/jart/gosip/sip"
 )
 
+func ReceiveMessagesFromAddr(sock net.Conn, c chan<- *sip.Msg, e chan<- error, addr *net.UDPAddr) {
+	buf := make([]byte, 2048)
+	var lhost string
+	var lport uint16
+	switch laddr := sock.LocalAddr().(type) {
+	case *net.UDPAddr:
+		lhost = laddr.IP.String()
+		lport = uint16(laddr.Port)
+	case *net.TCPAddr:
+		lhost = laddr.IP.String()
+		lport = uint16(laddr.Port)
+	}
+	for {
+		amt, err := sock.Read(buf)
+		if err != nil {
+			e <- err
+			break
+		}
+		ts := time.Now()
+		packet := buf[0:amt]
```

Comment:
This seems to imply that the entire SIP message will be contained in a single `Read`. That isn't necessarily the case though, is it? Is it not required to skip newlines before the start-line, then read all headers, then read `Content-Length` bytes of the body?